lib/src/config.cc

```diff
@@ -304,6 +311,88 @@ namespace hocon {
+        } else {
+            throw bad_value_exception(*v->origin(), path, "Value at '" + path + "' was not an integer or string.");
+        }
+    }
+
+    double config::get_duration(string const& path, time_unit unit) const {
+        auto timespan = get_duration(path);
+        switch (unit) {
+            case time_unit::MILLISECONDS:
+                return timespan.count();
+            case time_unit::SECONDS:
+                return timespan.count() / 1000.0;
+            case time_unit::MINUTES:
+                return timespan.count() / 60000.0;
+            case time_unit::HOURS:
+                return timespan.count() / 3600000.0;
+            case time_unit::DAYS:
+                return timespan.count() / 86400000.0;
+            default:
+                throw config_exception("Not a valid time_unit");
+        }
+    }
+
+    milliseconds config::parse_duration(string input, shared_origin origin_for_exception, string path_for_exception) {
+        boost::algorithm::trim(input);
+        string original_unit_string = get_units(input);
+        string unit_string = original_unit_string;
+        string number_string = boost::algorithm::trim_copy(input.substr(0, input.length() - unit_string.length()));
+
+        if (number_string.empty()) {
+            throw bad_value_exception(*origin_for_exception, path_for_exception, "No number in duration value '" + input + "'");
+        }
+
+        if (unit_string.length() > 2 && unit_string.back() != 's') {
+            unit_string += "s";
+        }
+
+        double number = 0;
+        try {
+            number = boost::lexical_cast<double>(number_string);
+        } catch (boost::bad_lexical_cast& ex) {
+            throw bad_value_exception(*origin_for_exception, path_for_exception, "Value '" + number_string + "' could not be converted to a number.");
+        }
+        if (unit_string.empty() || unit_string == "ms" || unit_string == "millis" || unit_string == "milliseconds") {
+            return milliseconds(static_cast<int64_t>(number));
+        } else if (unit_string == "d" || unit_string == "days") {
+            // boost::chrono::duration does not support conversion from doubles to ints very well
+            int64_t millis = static_cast<int64_t>(number * 86400000);
+            return milliseconds(millis);
+        } else if (unit_string == "s" || unit_string == "seconds") {
+            int64_t millis = static_cast<int64_t>(number * 1000);
+            return milliseconds(millis);
+        } else if (unit_string == "h" || unit_string == "hours") {
+            int64_t millis = static_cast<int64_t>(number * 3600000);
+            return milliseconds(millis);
+        } else if (unit_string == "m" || unit_string == "minutes") {
+            int64_t millis = static_cast<int64_t>(number * 60000);
+            return milliseconds(millis);
+        } else {
+            throw bad_value_exception(*origin_for_exception, path_for_exception,
```

Comment:
I don't think boost::chrono is the correct equivalent to java.time.Duration. boost::chrono is more about representing program execution time (clock cycles), not physical time. I'd look at boost::posix_time::duration instead. The boost::date_time library has been [proposed](http://www.crystalclearsoftware.com/libraries/date_time/proposal_v75/date_time_standard.html) for inclusion in the standard, but I don't think it's scheduled.

posix_time provides a 96-bit nanosecond resolution option, and facilities for translating between different units.